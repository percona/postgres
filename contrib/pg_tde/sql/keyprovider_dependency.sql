CREATE EXTENSION IF NOT EXISTS pg_tde;

SELECT pg_tde_add_key_provider_file('mk-file','/tmp/pg_tde_test_keyring.per');
SELECT pg_tde_add_key_provider_file('free-file','/tmp/pg_tde_test_keyring_2.per');
SELECT pg_tde_add_key_provider_vault_v2('V2-vault','vault-token','percona.com/vault-v2/percona','/mount/dev','ca-cert-auth');

SELECT * FROM pg_tde_list_all_key_providers();

SELECT pg_tde_set_principal_key('test-db-principal-key','mk-file');

DROP EXTENSION pg_tde;
