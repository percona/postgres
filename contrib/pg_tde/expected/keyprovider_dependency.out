CREATE EXTENSION IF NOT EXISTS pg_tde;
SELECT pg_tde_add_key_provider_file('mk-file','/tmp/pg_tde_test_keyring.per');
 pg_tde_add_key_provider_file 
------------------------------
                            1
(1 row)

SELECT pg_tde_add_key_provider_file('free-file','/tmp/pg_tde_test_keyring_2.per');
 pg_tde_add_key_provider_file 
------------------------------
                            2
(1 row)

SELECT pg_tde_add_key_provider_vault_v2('V2-vault','vault-token','percona.com/vault-v2/percona','/mount/dev','ca-cert-auth');
 pg_tde_add_key_provider_vault_v2 
----------------------------------
                                3
(1 row)

SELECT * FROM pg_tde_list_all_key_providers();
 id | provider_name | provider_type |                                                                    options                                                                    
----+---------------+---------------+-----------------------------------------------------------------------------------------------------------------------------------------------
  1 | mk-file       | file          | {"type" : "file", "path" : "/tmp/pg_tde_test_keyring.per"}
  2 | free-file     | file          | {"type" : "file", "path" : "/tmp/pg_tde_test_keyring_2.per"}
  3 | V2-vault      | vault-v2      | {"type" : "vault-v2", "url" : "percona.com/vault-v2/percona", "token" : "vault-token", "mountPath" : "/mount/dev", "caPath" : "ca-cert-auth"}
(3 rows)

SELECT pg_tde_set_principal_key('test-db-principal-key','mk-file');
 pg_tde_set_principal_key 
--------------------------
 t
(1 row)

DROP EXTENSION pg_tde;
